File: educational-system-consumer-student-9001/src/main/java/edu/hubu/consumerstudent/service/IStudentSelectCourseService.java
package edu.hubu.consumerstudent.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.student.StudentSelectCourse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "student-provider-service")
public interface IStudentSelectCourseService {

    @GetMapping("/studentselectCourse/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/studentselectCourse/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "studentselectCourseId") String orderBy);

    @GetMapping("/studentselectCourse/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "studentselectCourseId") String orderBy
    );

    @PostMapping("/studentselectCourse/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "studentselectCourseId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.StudentSelectCourse studentselectCourse
    );

    @PostMapping("/studentselectCourse/")
    CommonResult save(@RequestBody StudentSelectCourse studentselectCourse);

    @PutMapping("/studentselectCourse/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody StudentSelectCourse studentselectCourse);


    @DeleteMapping("/studentselectCourse/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/student/IStudentSelectCourseService.kt
package edu.hubu.commons.service.student

import edu.hubu.commons.model.student.StudentSelectCourse;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface IStudentSelectCourseService {
    fun save(t: StudentSelectCourse): StudentSelectCourse
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: StudentSelectCourse): StudentSelectCourse
    fun selectOneById(id: String): StudentSelectCourse
    fun selectAllOrderBy(order: String = "studentselectCourseId"): List<StudentSelectCourse>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "studentselectCourseId"): List<StudentSelectCourse>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "studentselectCourseId", t: edu.hubu.commons.mybatis.model.StudentSelectCourse): List<edu.hubu.commons.mybatis.model.StudentSelectCourse>
}
